### ISSUE TITLE
Postgresql release upgrade impossible when using auth.existingSecret

### ISSUE THREAD

**Issue by sebastien-prudhomme** [bitnami/charts#21339]
### Name and Version

bitnami/postgresql 13.2.21

### What architecture are you using?

amd64

### What steps will reproduce the bug?

When trying to upgrade a postgresql release when using `auth.existingSecret`, I got an error message.

### Are you using any custom parameters or values?

values.yaml:

```yaml
architecture: standalone
auth:
  database: myapp
  existingSecret: myapp-postgresql
  username: myapp
primary:
  persistence:
    size: 1Gi
  resources:
    limits: {}
    requests:
      cpu: 250m
      memory: 256Mi
serviceAccount:
  create: true
```

Existing secret:

```yaml
apiVersion: v1
kind: Secret
metadata:
  labels:
    app: myapp-postgresql
  name: myapp-postgresql
  namespace: myapp
type: Opaque
data:
  password: ****
  postgres-password: ****
```

### What do you see instead?

Error message:

```
  Error: Failed to render chart: exit status 1: Error: execution error at (postgresql/templates/secrets.yaml:23:16):
  PASSWORDS ERROR: You must provide your current passwords when upgrading the release.
                   Note that even after reinstallation, old credentials may be needed as they may be kept in persistent volume claims.
                   Further information can be obtained at https://docs.bitnami.com/general/how-to/troubleshoot-helm-chart-issues/#credential-errors-while-upgrading-chart-releases
      'global.postgresql.auth.password' must not be empty, please add '--set global.postgresql.auth.password=$PASSWORD' to the command. To get the current value:
`````

### Additional information

I've also notice that when installing for the first time the chart, I've got this wrong message in the NOTES:

```
WARNING: PostgreSQL has been configured without authentication, this is not recommended for production environments.
```

**Comment by carrodher** [comment]
Are you able to follow the notes that are shown after the error?

**Comment by sebastien-prudhomme** [comment]
@carrodher I don't want to put the password in a value, I want to use an existing secret to get the password. Upgrade is working fine if I use a value for the password.

I will try to propose a fix.

**Comment by sebastien-prudhomme** [comment]
The problem is related to the `helmfile` tool I'm using and the fact that the chart is using `lookup` in templates to detect if a secret is yet created.

Unfortunetely it does't work yet. It should work when the helm-diff plugin will implement the `--dry-run=server` available since helm 3.13.
